query-graphs-js/src/pathTree.ts

```diff
@@ -48,6 +48,8 @@ export class PathTree<TTrigger, RV extends Vertex = Vertex, TNullEdge extends nu
     readonly localSelections: readonly SelectionSet[] | undefined,
     private readonly triggerEquality: (t1: TTrigger, t2: TTrigger) => boolean,
     private readonly childs: Child<TTrigger, Vertex, TNullEdge>[],
+    readonly contextToSelection: Map<string, SelectionSet> | null,
+    public parameterToContext: Map<string, ContextAtUsageEntry> | null,
```

Comment:
After looking at `computeGroupsForTree()` and thinking a bit more, I think it'll be easier if you also move `contextToSelection` to be in `Child` as well (and change elements()` and `childElements()` to also give these two additional fields, which will let you access them more cleanly in `computeGroupsForTree()`. I think you'd also be fine storing a set of context IDs instead of a `Map` as `contextToSelection` (I don't think you explicitly use the selection sets, at least).